src/vhost.c

```diff
@@ -389,17 +304,16 @@ struct pg_brick *pg_vhost_new(const char *name, uint64_t flags,
 
 static void vhost_destroy(struct pg_brick *brick, struct pg_error **errp)
 {
-	struct pg_vhost_state *state;
-
-	state = pg_brick_get_state(brick, struct pg_vhost_state);
+	struct pg_vhost_state *state =
+		pg_brick_get_state(brick, struct pg_vhost_state);
 	rte_vhost_driver_unregister(state->socket->path);
-
 	pthread_mutex_lock(&mutex);
-	g_remove(state->socket->path);
-	LIST_REMOVE(state->socket, socket_list);
 	g_free(state->socket->path);
 	g_free(state->socket);
-
+	LIST_REMOVE(state->socket, socket_list);
```

Comment:
I'm very sceptical about changing the order of those 2 operation,
I mean here you have at first doing a 
```
g_free(state->socket->path);
g_free(state->socket);
```
and then you access them with LIST_REMOVE and g_remove

Reply:
yeah that's dirty... fixed